trusted-services.mk

```diff
@@ -0,0 +1,50 @@
+################################################################################
+# Paths to Trusted Services source and output
+################################################################################
+TS_PATH			?= $(ROOT)/trusted-services
+TS_BUILD_PATH		?= $(OUT_PATH)/ts-build
+TS_INSTALL_PREFIX	?= $(OUT_PATH)/ts-install
+
+################################################################################
+# Secure Partitions
+################################################################################
+.PHONY: ffa-sp-all ffa-sp-all-clean ffa-sp-all-realclean
+
+optee-os-common: ffa-sp-all
+optee-os-clean: ffa-sp-all-clean
+
+ffa-sp-all-realclean:
+	rm -rf $(TS_INSTALL_PREFIX)/opteesp
+
+# Helper macro to build and install Trusted Services Secure Partitions (SPs).
+# Invokes CMake to configure, build and install the SP, adds the SP output image
+# to the SP_PATH list of optee_os and complies the SP manifest dts to dtb.
+#
+# Parameter list:
+# 1 - SP deployment name (e.g. internal-trusted-storage, crypto, etc.)
+# 2 - SP canonical UUID (e.g. dc1eef48-b17a-4ccf-ac8b-dfcff7711b14)
+# 3 - SP additional build flags (e.g. -DTS_PLATFORM=<...>)
+define build-sp
+.PHONY: ffa-$1-sp
+ffa-$1-sp:
+	CROSS_COMPILE=$(AARCH64_CROSS_COMPILE) cmake -G"Unix Makefiles" \
+		-S $(TS_PATH)/deployments/$1/opteesp -B $(TS_BUILD_PATH)/$1 \
+		-DCMAKE_INSTALL_PREFIX=$(TS_INSTALL_PREFIX) $(SP_COMMON_FLAGS) $3
+	cmake --build $(TS_BUILD_PATH)/$1
```

Comment:
Replaced with `$(MAKE)`, probably this is the cleanest solution, and the `cmake --build` wrapper doesn't get in the way.
Note: since we're inside a `define` here, we have to use `$$(MAKE)`.

Reply:
More information about passing parallel build related parameters to sub-`make` is available at
https://www.gnu.org/software/make/manual/make.html#Error-Messages (see explanation of `warning: jobserver unavailable ...`) and https://www.gnu.org/software/make/manual/make.html#MAKE-Variable

Reply:
Thanks, perfect!